tests/integration/app_3000/tests.ts

```diff
@@ -42,118 +27,108 @@ const webServer = new Drash.Http.Server({
-);
 
-// Set up the events
+////////////////////////////////////////////////////////////////////////////////
+// TESTS ///////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
 
-socketServer.openChannel("chan1");
-socketServer.on(
-  "chan1",
-  ((packet: any) => {
-    storage["chan1"].messages.push(packet.message);
-  }),
+console.log(
+  "\nIntegration tests: testing different channels can be opened and work.\n",
 );
 
-Deno.test("chan1 should exist", () => {
-  assertEquals("chan1", socketServer.getChannel("chan1").name);
-});
-
-Deno.test("chan2 should exist again", () => {
-  socketServer.openChannel("chan2");
-  socketServer.on(
-    "chan2",
-    ((packet: any) => {
-      storage["chan2"].messages.push(packet.message);
-    }),
-  );
-  assertEquals("chan2", socketServer.getChannel("chan2").name);
-});
-
-Deno.test("chan1 should have 1 message", async () => {
-  await sendMessage("chan1", "This is a chan1 message.");
-  assertEquals(
-    storage["chan1"].messages,
-    [
-      "This is a chan1 message.",
-    ],
-  );
-});
+let storage: any = {
+  "chan1": {
+    messages: [],
+  },
+  "chan2": {
+    messages: [],
+  },
+};
 
-Deno.test("chan1 should have 2 messages", async () => {
-  await sendMessage("chan1", "This is a chan1 message #2.");
-  assertEquals(
-    storage["chan1"].messages,
-    [
-      "This is a chan1 message.",
-      "This is a chan1 message #2.",
-    ],
-  );
-});
+Rhum.testPlan("app_3000", () => {
+  socketServer.openChannel("chan1");
 
-Deno.test("chan2 should have 1 message", async () => {
-  await sendMessage("chan2", "This is a chan2 message.");
-  assertEquals(
-    storage["chan2"].messages,
-    [
-      "This is a chan2 message.",
-    ],
-  );
-});
+  socketServer.on("chan1", (packet: Packet) => {
+    storage["chan1"].messages.push(packet.message);
+  });
 
-Deno.test("chan2 should be closed", () => {
-  socketServer.closeChannel("chan2");
-  assertEquals(undefined, socketServer.getChannel("chan2"));
+  Rhum.testSuite("channel 1", () => {
+    Rhum.testCase("chan1 should exist", () => {
+      Rhum.asserts.assertEquals("chan1", socketServer.getChannel("chan1").name);
+    });
+    Rhum.testCase("chan1 should have a message", async () => {
+      await sendMessage(JSON.stringify({
+        data: {
+          send_packet: {
+            to: "chan1",
+            message: "chan message 1-1",
+          },
+        },
+      }));
+      Rhum.asserts.assertEquals(
+        storage["chan1"].messages,
+        ["chan message 1-1"],
+      );
+    });
+    Rhum.testCase("chan2 should have a message", async () => {
+      socketServer.openChannel("chan2");
+      socketServer.on("chan2", () => {
+        socketServer.on("chan2", (packet: Packet) => {
+          storage["chan2"].messages.push(packet.message);
+        });
+      });
```

Comment:
Was just curious why there's 2 `on`'s for the same channel?

Reply:
woops LOL

Reply:
b57aa3a